src/app/pages/OnDemandTvPage/VideoPlayer/index.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useContext } from 'react';
+import { useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+import {
+  GEL_SPACING,
+  GEL_SPACING_DBL,
+  GEL_SPACING_QUAD,
+} from '@bbc/gel-foundations/spacings';
+import { GEL_GROUP_2_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';
+import { string, bool } from 'prop-types';
+import {
+  CanonicalMediaPlayer,
+  AmpMediaPlayer,
+  MediaMessage,
+} from '@bbc/psammead-media-player';
+import pathOr from 'ramda/src/pathOr';
+import { RequestContext } from '#contexts/RequestContext';
+import { ServiceContext } from '#contexts/ServiceContext';
+import getEmbedUrl from '#lib/utilities/getEmbedUrl';
+import getPlaceholderImageUrl from '../../../routes/utils/getPlaceholderImageUrl';
+
+const VideoPlayerWrapper = styled.div`
+  width: calc(100% + ${GEL_SPACING_DBL});
+  margin: 0 -${GEL_SPACING};
+  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
+    width: calc(100% + ${GEL_SPACING_QUAD});
+    margin: 0 -${GEL_SPACING_DBL};
+  }
+`;
+
+const landscapeRatio = '56.25%';
+const MediaMessageWrapper = styled.div`
+  padding-top: ${landscapeRatio};
+  position: relative;
+`;
+
+const VideoPlayer = ({ assetId, masterBrand, imageUrl, isExpired }) => {
+  const { lang, translations, service } = useContext(ServiceContext);
+  const { isAmp, platform } = useContext(RequestContext);
+  const location = useLocation();
+  const isValidPlatform = ['amp', 'canonical'].includes(platform);
+  const mediaInfo = {
+    title: 'On-demand TV',
+    type: 'video',
+  };
+  const noJsMessage = pathOr(
+    `This ${mediaInfo.type} cannot play in your browser. Please enable JavaScript or try a different browser.`,
+    ['media', 'noJs'],
+    translations,
+  );
+
+  if (isExpired) {
+    const expiredContentMessage = pathOr(
+      'This content is no longer available',
+      ['media', 'contentExpired'],
+      translations,
+    );
+
+    return (
+      <MediaMessageWrapper>
+        <MediaMessage service={service} message={expiredContentMessage} />
+      </MediaMessageWrapper>
+    );
+  }
+  const placeholderSrc = getPlaceholderImageUrl(imageUrl);
+
+  if (!isValidPlatform || !masterBrand || !assetId) return null;
+
+  const mediaId = `${service}/${masterBrand}/${assetId}/${lang}`;
+
+  const embedUrl = getEmbedUrl({
+    mediaId,
+    type: 'media',
+    isAmp,
+    queryString: location.search,
+  });
+
+  const iframeTitle = pathOr(
+    'Video player',
+    ['mediaAssetPage', 'videoPlayer'],
+    translations,
+  );
+
+  return (
+    <VideoPlayerWrapper>
```

Comment:
I can't help but think is there not **some** way to refactor both this and `CpsAssetMediaPlayer` / `MediaPlayer` to inherit as much of the same shared code as possible? An intermediary component like `CpsAssetMediaPlayer` (but renamed... sheesh I've always hated that component name) could handle blocks transformation and pass down props compatible with both 'kinds' of media player?

Reply:
Yep - I really wanted to try to reuse the on demand audio player for this, but there were just too many differences

Probably not relevant to this, but I think we need to consolidate our media player components eventually - we have too many

- `MediaPlayer` is too tightly coupled to Optimo's data format
- `CPSMediaPlayer` is just a hack to allow CPS pages to reuse the `MediaPlayer` and work around the tight coupling
- `CPSMediaPlayer` is also used for TC2 assets
- `AudioPlayer` and `VideoPlayer` are pretty reasonable I think, but they're not really reusable in their current form

Reply:
First we'd want to refactor the getInitialData for MAPs etc to match the data format for liveRadio, OD Radio & OD TV. This would make it substantially easier to use the same components.

Reply:
Please create a ticket to refactor the existing media players and add it to the backlog.

Reply:
Nice one will do, this definitely feels like its own stream of work. I do think it's important - this isn't just vanity tech debt. Maintaining the two different components could get costly and clunky, you can already see a tiny impact of it in my comment on the the styling of this player being correct for mobile, but the MAP player not.

Agreed though that any 'homogenising' of the existing media player suite should not be to the detriment of their simplicity. Making reusable components that are so complex no one likes reusing them is useless.

Also shout out to my absolute favourite piece in all of this:

> CPSMediaPlayer is also used for TC2 assets